tests/typ/compiler/raw.typ

```diff
@@ -0,0 +1,145 @@
+// Test new raw parser
+// Ref: false
+
+---
+#let empty = (
+  name: "empty",
+  input: ``,
+  text: "",
+)
+
+#let backtick = (
+  name: "backtick",
+  input: ``` ` ```,
+  text: "`",
+  block: false,
+)
+
+#let lang-backtick = (
+  name: "lang-backtick",
+  input: ```js ` ```,
+  lang: "js",
+  text: "`",
+  block: false,
+)
+
+#let lang-space = (
+  name: "lang-space",
+  input: ```js test ```,
+  lang: "js",
+  text: "test ",
+  block: false,
+)
+
+#let lang-newline = (
+  name: "lang-newline",
+  input: ```js
+test
+```,
+  lang: "js",
+  text: "test",
+  block: true,
+)
+
+#let blocky = (
+  name: "blocky",
+  input: {
+```
+test
+```
+},
+  text: "test",
+  block: true,
+)
+
+#let blocky-dedent = (
+  name: "blocky-dedent",
+  input: {
+```
+ test
+ ```
+  },
+  text: "test",
+  block: true,
+)
+
+#let blocky-dedent-dont-considering-first-line = (
+  name: "blocky-dedent-dont-considering-first-line",
+  input: {
+```      
+ test
+ ```
+  },
+  text: "test",
+  block: true,
+)
+
+#let blocky-dedent-lastline = (
+  name: "blocky-dedent-lastline",
+  input: ```
+  test
+ ```,
+  text: " test",
+  block: true,
+)
+
+#let blocky-dedent-lastline2 = (
+  name: "blocky-dedent-lastline2",
+  input: ```
+  test
+   ```,
+  text: "test",
+  block: true,
+)
+
+#let blocky-tab = (
+  name: "blocky-tab",
+  input: {
+```
+	test
+```
+},
+  text: "\ttest",
+  block: true,
+)
+
+#let blocky-tab-dedent = (
+  name: "blocky-tab-dedent",
+  input: {
+```
+	test
+  
+ ```
+},
+  text: "test\n ",
+  block: true,
+)
+
+
```

Comment:
Please remove double newlines. I think we can also skip the main function and have its contents top-level.